boa/src/builtins/uri/mod.rs

```diff
@@ -0,0 +1,108 @@
+//! This module implements the global `decodeURI` and encodURI functions.
+
+use std::borrow::Borrow;
+
+use crate::value::RcString;
+use crate::{
+    builtins::BuiltIn, object::FunctionBuilder, property::Attribute, value::Value, BoaProfiler,
+    Context, Result,
+};
+use percent_encoding::{percent_decode, utf8_percent_encode, AsciiSet, CONTROLS};
+
+type EncodeFuncType = fn(&RcString) -> Value;
+
+// https://url.spec.whatwg.org/#fragment-percent-encode-set
+const ENCODE_FRAGMENT: &AsciiSet = &CONTROLS.add(b' ').add(b'"').add(b'<').add(b'>').add(b'`');
+
+fn decode(str: &RcString) -> Value {
+    Value::string(
+        percent_decode(str.as_bytes())
+            .decode_utf8()
+            .unwrap()
+            .borrow(),
+    )
+}
+
+fn encode(str: &RcString) -> Value {
+    Value::string(utf8_percent_encode(str, ENCODE_FRAGMENT).to_string())
+}
+
+#[derive(Debug, Clone, Copy)]
+pub(crate) struct Uri;
+
+impl BuiltIn for Uri {
+    const NAME: &'static str = "Uri";
+
+    fn attribute() -> Attribute {
+        Attribute::WRITABLE | Attribute::NON_ENUMERABLE | Attribute::CONFIGURABLE
+    }
+
+    fn init(context: &mut Context) -> (&'static str, Value, Attribute) {
+        let _timer = BoaProfiler::global().start_event(Self::NAME, "init");
+
+        let decode_uri = FunctionBuilder::new(context, Self::decode_uri)
+            .name("decodeURI")
+            .length(1)
+            .callable(true)
+            .constructable(false)
+            .build();
+
+        let encode_uri = FunctionBuilder::new(context, Self::encode_uri)
+            .name("encodeURI")
+            .length(1)
+            .callable(true)
+            .constructable(false)
+            .build();
+
+        context.register_global_property("decodeURI", decode_uri, Attribute::all());
+        context.register_global_property("encodeURI", encode_uri, Attribute::all());
+
+        let global = context.global_object();
+
+        (Self::NAME, Value::undefined(), Self::attribute())
+    }
+}
+
+impl Uri {
+    pub(crate) const LENGTH: usize = 1;
+
+    pub(crate) fn handle_uri(args: &[Value], cb: EncodeFuncType) -> Result<Value> {
+        let first_arg = match args.get(0) {
```

Comment:
Since the `get()` function returns an `Option`, you can call it's methods to make the code easier to read, by using, for example, `map()` and `unwrap_or_default()`.